robot/web/web_c.go

```diff
@@ -23,10 +23,19 @@ func (svc *webService) Reconfigure(ctx context.Context, _ resource.Dependencies,
 	if !svc.isRunning {
 		return nil
 	}
-	return svc.streamServer.AddNewStreams(svc.cancelCtx)
+
+	svc.streamServerInitCloseMu.Lock()
+	defer svc.streamServerInitCloseMu.Unlock()
+	// May be nil if this is being run after closeStreamServer.
+	if svc.streamServer != nil {
+		return svc.streamServer.AddNewStreams(svc.cancelCtx)
```

Comment:
should streamServer ever be nil if the web server is up? feels like that'd be bad since that would be we won't be able to get video streams

Reply:
a test would be to get the web server in such a state that the streamServer is nil during a reconfigure and see if we can still access webrtc video streams (UI Live camera streams, for example)